Apps/Admin/Server/Services/BetaFeatureAccessService.cs

```diff
@@ -0,0 +1,92 @@
+
+    /// <inheritdoc/>
+    /// <param name="userProfileDelegate">The user profile delegate to interact with the DB.</param>
+    /// <param name="betaFeatureAccessDelegate">The beta feature access delegate to interact with the DB.</param>
+    /// <param name="mappingService">The injected mapping service.</param>
+    /// <param name="logger">The injected logger.</param>
+    public class BetaFeatureAccessService(
+        IUserProfileDelegate userProfileDelegate,
+        IBetaFeatureAccessDelegate betaFeatureAccessDelegate,
+        IAdminServerMappingService mappingService,
+        ILogger<BetaFeatureAccessService> logger)
+        : IBetaFeatureAccessService
+    {
+        /// <inheritdoc/>
+        public async Task SetUserAccessAsync(string email, IList<BetaFeature> betaFeatures, CancellationToken ct = default)
+        {
+            logger.LogDebug("Email: {Email} - Beta Features: {Features}", email, betaFeatures);
+            IList<UserProfile> userProfiles = await userProfileDelegate.GetUserProfileAsync(email, true, ct);
+
+            if (userProfiles.Count == 0)
+            {
+                throw new NotFoundException(ErrorMessages.UserProfileNotFound);
+            }
+
+            IEnumerable<string> hdids = userProfiles.Select(x => x.HdId);
+            IEnumerable<Database.Models.BetaFeatureAccess> existingBetaFeatures = await betaFeatureAccessDelegate.GetAsync(hdids, ct);
+
+            IEnumerable<Database.Models.BetaFeatureAccess> betaFeaturesToDelete = existingBetaFeatures
+                .Where(x => !betaFeatures.Contains(mappingService.MapToBetaFeature(x.BetaFeatureCode)));
+
+            IEnumerable<Database.Models.BetaFeatureAccess> betaFeaturesToAdd = betaFeatures
+                .Where(betaFeature => existingBetaFeatures.All(betaFeatureAccess => betaFeatureAccess.BetaFeatureCode != mappingService.MapToBetaFeature(betaFeature)))
+                .SelectMany(betaFeature => userProfiles.Select(profile => mappingService.MapToBetaFeatureAccess(profile.HdId, betaFeature)));
+
+            await betaFeatureAccessDelegate.DeleteRangeAsync(betaFeaturesToDelete, false, ct);
+            await betaFeatureAccessDelegate.AddRangeAsync(betaFeaturesToAdd, true, ct);
+        }
+
+        /// <inheritdoc/>
+        public async Task<IEnumerable<BetaFeature>> GetUserAccessAsync(string email, CancellationToken ct = default)
+        {
+            IList<UserProfile> userProfiles = await userProfileDelegate.GetUserProfileAsync(email, true, ct);
+
+            if (userProfiles.Count == 0)
+            {
+                throw new NotFoundException(ErrorMessages.UserProfileNotFound);
+            }
+
+            ICollection<BetaFeatureCode> betaFeatureCodes = userProfiles
+                .SelectMany(profile => profile.BetaFeatureCodes)
+                .Distinct()
+                .ToList();
+
+            return betaFeatureCodes.Select(x => mappingService.MapToBetaFeature(x.Code));
```

Comment:
The ToList() is not serving any purpose where it is.  It could be moved after the last Select().